src/main/kotlin/org/opensearch/commons/notifications/model/SNS.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.commons.notifications.model
+
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.notifications.NotificationConstants.ROLE_ARN_FIELD
+import org.opensearch.commons.notifications.NotificationConstants.TOPIC_ARN_FIELD
+import org.opensearch.commons.utils.fieldIfNotNull
+import org.opensearch.commons.utils.logger
+import java.io.IOException
+import java.util.regex.Pattern
+
+/**
+ * SNS notification data model
+ */
+data class SNS(val topicARN: String, val roleARN: String?) : BaseConfigData {
+
+    init {
+        require(SNS_ARN_REGEX.matcher(topicARN).find()) { "Invalid AWS SNS topic ARN: $topicARN" }
+        if (roleARN != null) {
+            require(IAM_ARN_REGEX.matcher(roleARN).find()) { "Invalid AWS role ARN: $roleARN " }
```

Comment:
Probably this can be moved to utils as this is required for SES as well

Reply:
moved